nutils/evaluable.py

```diff
@@ -2910,6 +3066,15 @@ def _jointdtype(*dtypes):
 def _gatherblocks(blocks):
   return tuple((ind, util.sum(funcs)) for ind, funcs in util.gather(blocks))
 
+def _gathersparsechunks(chunks):
+  chunks = tuple(chunks)
+  if not chunks:
+    return chunks
+  ndim = len(chunks[0])-1
+  if ndim == 0:
+    return chunks
```

Comment:
All indices have exactly the same shape as the values. Gathering the indices guarantees that the values in the `util.sum` have the same shape. Except if the dimension of the result is zero, in which case there are no indices and gathering is done on the empty tuple. You could argue that it doesn't make sense to return a chunk with shape other than `()`, because the result is summed anyway, but that is currently not the case. Shall we enforce this (`values.shape == () if ndim == 0`)?

Reply:
Ok I see the issue now with `util.sum`. Indeed that restriction makes sense to me, unless that leads to special case code all over the board. Otherwise we could simply handle it here as `return ((), util.sum(_sumall(func) for (func,) in chunks)` where `_sumall` sums all axes.

Reply:
It's relatively straightforward to impose this condition.